### ISSUE TITLE
[3.5] Rotating TextureRect's rotation will be messed up if the parent gets resized

### ISSUE THREAD

**Issue by Guusggg** [godotengine/godot#78275]
### Godot version

3.5.2.stable.mono.arch_linux

### System information

Manjaro Linux - 3.5.2.stable.mono.arch_linux - GLES3

### Issue description

Lately I'm doing more and more UI projects in Godot because it's so nice to make scalable UI's. This time, I have an icon (a sword) which is rotated around a specific pivot (the sword's handle). 

### Steps to reproduce

I use a TextureRect for this, with Keep Aspect Covered.

![image](https://github.com/godotengine/godot/assets/6216482/b4d3fa96-4ce1-4dcd-b0d0-6625d9bbb32f)

When I rotate the sword right now, it's perfectly fine.

![image](https://github.com/godotengine/godot/assets/6216482/6acfed80-f235-46a1-81b2-8f78b075dd71)

But now when the sword is rotated, and the sword's parent is resized, the rotation will be messed up.

![image](https://github.com/godotengine/godot/assets/6216482/d9eb4de9-0207-4900-bbb8-0f3aab31d43e)

When I look at the pivot, it is still at the same exact point as when it was when it was sized as 32x32. 
Is there anything I can do to fix it?

I used other Control's as "anchors" (parenting to another Control and use that position as center), but then I lose the scaling of the Sword. I don't really understand why, while everything will correctly increase in size in a Control, the pivot does not move with it.

### Minimal reproduction project
![image](https://github.com/godotengine/godot/assets/6216482/aef90110-5d83-4d43-a177-8a43674d392f)

[RotationExample.zip](https://github.com/godotengine/godot/files/11768146/RotationExample.zip)

**Comment by Sauermann** [comment]
This bug is not reproducible in Godot 4.

**Comment by felaugmar** [comment]
@Guusggg 

I've tested what you said and it's working as expected without messing with the `TextureRect`'s position.
The only difference is that I used the Windows, but I don't think that's the reason.

Could you upload a Minimal reproduction project?

**Comment by Guusggg** [comment]
@Sauermann I didn't mean to cause confusion, but my project is in Godot 3

@felaugmar I replicated a tiny example of the behaviour in the following project: 
[RotationExample.zip](https://github.com/godotengine/godot/files/11768146/RotationExample.zip). As you can see I placed the pivot in the center of the `TextureRect`, but it's not keeping it in the center. I hope this is clear enough; if not, please let me know!

**Comment by Sauermann** [comment]
> I didn't mean to cause confusion, but my project is in Godot 3

No worries. I just wanted to pinpoint the version.

**Comment by felaugmar** [comment]
@Guusggg 

> I replicated a tiny example of the behaviour in the following project: [RotationExample.zip](https://github.com/godotengine/godot/files/11768146/RotationExample.zip). As you can see I placed the pivot in the center of the `TextureRect`, but it's not keeping it in the center. What I would hope to see is the `TextureRect` rotating correctly around it's pivot, even when it's initial size increases or decreases. I hope this is clear enough; if not, please let me know!

Thanks, I understand it now.

It's the expected behavior, when you change the size of the parent, **Keep Aspect Covered** makes your sword enlarge its size as well, then you should update the pivot as according.

I've updated your MRP with an example: [RotationExample.zip](https://github.com/godotengine/godot/files/11770849/RotationExample.zip)

**Comment by Guusggg** [comment]
@felaugmar Thank you very much for checking it out. I think what you fixed for me was setting the pivot in the animation, however I just used the animation as an example of enlarging the parent control. Since this is expected behaviour, I assume there's just no other way than including a little script that places the pivot on the correct position every frame? 

Ideally what I would like is a Control with TextureRect that can rotate in it's animation, while still allowing the parent Control to fit into whatever space it needs (which means it will adjust it's size, which in turn the pivot of the TextureRect has to be adjusted). (_Just as an example, I don't use it_) In Unity, this is very easy and for me it's a little more natural behaviour for the pivot to adjust itself according to the size of the UI element (since everything is already reacting to size changes). 

Thank you very much again for looking at it, and taking the time to check out the reproduction example! 

TLDR; just use a script?

**Comment by felaugmar** [comment]
> Since this is expected behaviour, I assume there's just no other way than including a little script that places the pivot on the correct position every frame?

You're right.

> In Unity, this is very easy and for me it's a little more natural behaviour for the pivot to adjust itself according to the size of the UI element (since everything is already reacting to size changes).

I also think it's more natural to the pivot updating to size changes (and people usually expect it at the start), but I also like the manual behavior, it gives more control.

> Thank you very much again for looking at it, and taking the time to check out the reproduction example!

You're welcome 😄

**Comment by Guusggg** [comment]
Thank you very much for your time.
